opm/core/props/satfunc/SatfuncConsistencyChecks.cpp

```diff
@@ -0,0 +1,552 @@
+/*
+  Copyright 2024 Equinor AS
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <config.h>
+
+#include <opm/core/props/satfunc/SatfuncConsistencyChecks.hpp>
+
+#include <opm/material/fluidmatrixinteractions/EclEpsScalingPoints.hpp>
+
+#include <algorithm>
+#include <array>
+#include <cassert>
+#include <cmath>
+#include <cstddef>
+#include <functional>
+#include <limits>
+#include <memory>
+#include <numeric>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include <fmt/format.h>
+
+// ===========================================================================
+// Public member functions for SatfuncConsistencyChecks Template
+// ===========================================================================
+
+template <typename Scalar>
+Opm::SatfuncConsistencyChecks<Scalar>::
+SatfuncConsistencyChecks(std::string_view  pointName,
+                         const std::size_t numSamplePoints)
+    : pointName_       { pointName }
+    , numSamplePoints_ { numSamplePoints }
+    , formatPointID_   { [](const std::size_t i) { return fmt::format("{}", i); } }
+{}
+
+template <typename Scalar>
+Opm::SatfuncConsistencyChecks<Scalar>::
+SatfuncConsistencyChecks(SatfuncConsistencyChecks&& rhs)
+    : pointName_        { std::move(rhs.pointName_) }
+    , numSamplePoints_  { rhs.numSamplePoints_ }
+    , formatPointID_    { std::move(rhs.formatPointID_) }
+    , startCheckValues_ { std::move(rhs.startCheckValues_) }
+    , violations_       { std::move(rhs.violations_) }
+    , battery_          { std::move(rhs.battery_) }
+{}
+
+template <typename Scalar>
+Opm::SatfuncConsistencyChecks<Scalar>&
+Opm::SatfuncConsistencyChecks<Scalar>::operator=(SatfuncConsistencyChecks&& rhs)
+{
+    this->pointName_        = std::move(rhs.pointName_);
+    this->numSamplePoints_  = rhs.numSamplePoints_;
+    this->formatPointID_    = std::move(rhs.formatPointID_);
+    this->startCheckValues_ = std::move(rhs.startCheckValues_);
+    this->violations_       = std::move(rhs.violations_);
+    this->battery_          = std::move(rhs.battery_);
+
+    this->urbg_.reset();
+
+    return *this;
+}
+
+template <typename Scalar>
+void Opm::SatfuncConsistencyChecks<Scalar>::resetCheckSet()
+{
+    this->startCheckValues_.clear();
+    this->startCheckValues_.push_back(0);
+
+    for (auto& violation : this->violations_) {
```

Comment:
consider ViolationSample::clear() / reset.

Reply:
> consider ViolationSample::clear() / reset.

Good idea.  I've pushed an update to that effect (using `clear()`).